mhs/common/mhs_common/workflow/asynchronous_reliable.py

```diff
@@ -1,26 +1,220 @@
+                                                          ack_requested=True,
+                                                          ack_soap_actor="urn:oasis:names:tc:ebxml-msg:actor:toPartyMSH",
+                                                          sync_reply=True)
 
+    @timing.time_function
     async def handle_outbound_message(self, from_asid: Optional[str],
                                       message_id: str, correlation_id: str, interaction_details: dict,
                                       payload: str,
-                                      work_description_object: Optional[wd.WorkDescription])\
+                                      wdo: Optional[wd.WorkDescription]) \
             -> Tuple[int, str, Optional[wd.WorkDescription]]:
-        raise NotImplementedError()
 
+        logger.info('0001', 'Entered async reliable workflow to handle outbound message')
+        if not wdo:
+            wdo = wd.create_new_work_description(self.persistence_store,
+                                                 message_id,
+                                                 workflow.ASYNC_RELIABLE,
+                                                 outbound_status=wd.MessageStatus.OUTBOUND_MESSAGE_RECEIVED
+                                                 )
+            await wdo.publish()
+
+        try:
+            details = await self._lookup_endpoint_details(interaction_details)
+            url = details[self.ENDPOINT_URL]
+            to_party_key = details[self.ENDPOINT_PARTY_KEY]
+            cpa_id = details[self.ENDPOINT_CPA_ID]
+        except Exception:
+            await wdo.set_outbound_status(wd.MessageStatus.OUTBOUND_MESSAGE_TRANSMISSION_FAILED)
+            return 500, 'Error obtaining outbound URL', None
+
+        error, http_headers, message = await self._serialize_outbound_message(message_id, correlation_id,
+                                                                              interaction_details,
+                                                                              payload, wdo, to_party_key, cpa_id)
+        if error:
+            await wdo.set_outbound_status(wd.MessageStatus.OUTBOUND_MESSAGE_TRANSMISSION_FAILED)
+            return error[0], error[1], None
+
+        reliability_details = await self._lookup_reliability_details(interaction_details)
+        retry_interval_xml_datetime = reliability_details[common_asynchronous.MHS_RETRY_INTERVAL]
+        try:
+            retry_interval = DateUtilities.convert_xml_date_time_format_to_seconds(retry_interval_xml_datetime)
+        except isoerror.ISO8601Error:
+            await wdo.set_outbound_status(wd.MessageStatus.OUTBOUND_MESSAGE_TRANSMISSION_FAILED)
+            return 500, 'Error when converting retry interval: {} to seconds'.format(retry_interval_xml_datetime), None
+
+        num_of_retries = reliability_details[common_asynchronous.MHS_RETRIES]
+
+        retries_remaining = num_of_retries
```

Comment:
this could just be declared inline 
retries_remaining = reliability_details[common_asynchronous.MHS_RETRIES]